go.mod

```diff
@@ -1,46 +1,47 @@
 module github.com/project-codeflare/codeflare-common
 
-go 1.21
+go 1.23.0
+
+toolchain go1.23.4
```

Comment:
Is there a reason to use a toolchain? Downstream Kueue uses 1.23.0.
https://github.com/opendatahub-io/kueue/blob/dev/go.mod#L3